aiokatcp/test/test_client.py

```diff
@@ -356,6 +350,27 @@ class TestClientNoReconnect(BaseTestClientAsync):
             await client_task
 
 
+@timelimit
+@istest
+class TestClientNoMidSupport(BaseTestClientAsync):
+    @timelimit(1)
+    async def setUp(self) -> None:
+        self.server, self.client_queue = await self.make_server(self.loop)
+        self.client, self.remote_reader, self.remote_writer = \
+            await self.make_client(self.server, self.client_queue)
+        self.addCleanup(self.remote_writer.close)
+
+    async def test_without_mid(self):
+        self.remote_writer.write(b'#version-connect katcp-protocol 5.0-M\n')
+        await self.client.wait_connected()
+        future = self.loop.create_task(self.client.request('echo'))
+        await self.check_received(re.compile(br'^\?echo\n\Z'))
+        await self.write(b'#echo an\\_inform\n')
+        await self.write(b'!echo ok\n')
+        result = await future
+        self.assertEqual(result, ([], [Message.inform('echo', b'an inform')]))
+
```

Comment:
Would it be possible to test that sending a second request while the first is busy actually blocks?  Sounds tricky to test.

Reply:
I've added a test.